packages/react-core/src/components/Slider/Slider.tsx

```diff
@@ -196,26 +207,34 @@ export const Slider: React.FunctionComponent<SliderProps> = ({
     const newPercentage = getPercentage(newPosition, end);
 
     thumbRef.current.style.setProperty('--pf-c-slider--value', `${newPercentage}%`);
-    setValue(newPercentage);
-
-    /* If discrete, snap to closest step value */
-    if (isDiscrete && steps) {
-      const stepIndex = steps.findIndex(step => step.value >= newPercentage);
-      if (steps[stepIndex].value === newPercentage) {
-        snapValue = steps[stepIndex].value;
+    // convert percentage to value
+    const newValue = (newPercentage * (max - min)) / 100 + min;
+    setValue(newValue);
+
+    /* If custom steps are discrete, snap to closest step value */
+    if (!areCustomStepsContinuous && customSteps) {
+      const stepIndex = customSteps.findIndex(stepObj => stepObj.value >= newPercentage);
+      if (customSteps[stepIndex].value === newPercentage) {
+        snapValue = customSteps[stepIndex].value;
       } else {
-        const midpoint = (steps[stepIndex].value + steps[stepIndex - 1].value) / 2;
+        const midpoint = (customSteps[stepIndex].value + customSteps[stepIndex - 1].value) / 2;
         if (midpoint > newPercentage) {
-          snapValue = steps[stepIndex - 1].value;
+          snapValue = customSteps[stepIndex - 1].value;
         } else {
-          snapValue = steps[stepIndex].value;
+          snapValue = customSteps[stepIndex].value;
         }
       }
+    } else if (!customSteps && step !== 1) {
+      // snap to new value if not custom steps and step is not 1
+      snapValue = Math.round((newValue - min) / step) * step + min;
+      // snapValue = Math.round(newValue / step) * step;
     }
 
-    // Call value change callback
-    if (onValueChange && !isDiscrete && !steps) {
-      onValueChange(newPercentage);
+    // Call value change callback unless custom steps are discrete or no custom steps and step is !== 1.  The value will snap after the thumb is released.
+    if (onChange) {
+      if ((areCustomStepsContinuous && customSteps) || (!customSteps && step !== 1)) {
```

Comment:
The `step !== 1` condition here is preventing the `onChange` from running if `step={1}` or the prop is omitted completely and it defaults to 1, which in the example has the effect of allowing the slider to be moved with the mouse but not updating the value displayed.  Note that using the keyboard still works, I believe because `handleKeyPressOnInput` does not have this same condition before triggering the `onChange`.